pilot/pkg/model/context.go

```diff
@@ -1122,3 +1122,44 @@ func OutboundListenerClass(t NodeType) istionetworking.ListenerClass {
 	}
 	return istionetworking.ListenerClassSidecarOutbound
 }
+
+// PreferIPv4FamilyForProxy help to confirm that a proxy IP family is IPv4 or IPv6 based on its k8s service addresses.
+// The proxy configuration should respect the k8s service addresses order, such as service with IP family [IPv4, IPv6] or [IPv6, IPv4].
```

Comment:
How can you check the prefer IP family based on service? If a pod doesn't have any service, or if a pod has multiple service with different preference of each service. Any why you need to decide the proxy's prefered IP family based on service? Those are totally two different level of configuration. I don't believe at proxy level, there is preference of IP family. For example, in openshift, it's always prefer ipv4 in dual stack mode.